File: src/fractal/world/time/_time.py
from __future__ import annotations
from typing import Iterator

from .step import Timestep


class Time(
    Iterator[Timestep],
):
    def __iter__(self) -> Time:
        return self

    def __next__(self) -> Timestep:
        pass

    def flow(self) -> Timestep:
        return next(self)

File: src/fractal/world/time/_time.rs
from __future__ import annotations
from typing import Iterator

from .step import Timestep


pub struct Time(
    Iterator<Timestep>,
):
    fn __iter__(&self) -> Time:
        return &self

    fn __next__(&self) -> Timestep:
        pass

    fn flow(&self) -> Timestep:
        return next(&self)